### ISSUE TITLE
The loss of the CPPD recognition algorithm does not decrease and fails to converge.I would like to ask what the possible reason might be. I look forward to your reply.

### ISSUE THREAD

**Issue by zhuzhiqiang148** [Topdu/OpenOCR#134]

**Comment by Topdu** [comment]
What is your configuration file like?

**Comment by zhuzhiqiang148** [comment]
> What is your configuration file like?

Global:
  device: gpu
  epoch_num: 500
  log_smooth_window: 20
  print_batch_step: 10
  output_dir: ./output/rec/data_cepen_3w/svtr_base_cppd/
  save_epoch_step: [15, 1]
  # evaluation is run every 2000 iterations
  eval_batch_step: [0, 500]
  eval_epoch_step: [0, 1]
  cal_metric_during_train: True
  pretrained_model: E:\Proj\PyProj\OpenOCR-main\openocr_repsvtr_ch.pth
  checkpoints:
  use_tensorboard: false
  infer_img:
  # for data or label process
  character_dict_path: &character_dict_path E:\Proj\PyProj\OpenOCR-main\tools\utils\cepen.txt
  # ./tools/utils/EN_symbol_dict.txt # 96en
  # ./tools/utils/ppocr_keys_v1.txt  # ch
  max_text_length: &max_text_length 25
  use_space_char: &use_space_char False
  save_res_path: ./output/rec/data_cepen_3w/predicts_svtr_base_cppd.txt
  use_amp: True

Optimizer:
  name: AdamW
  lr: 0.00065 # for 4gpus bs256/gpu
  weight_decay: 0.05
  filter_bias_and_bn: True

LRScheduler:
  name: OneCycleLR
  warmup_epoch: 1.5 # pct_start 0.075*20 = 1.5ep
  cycle_momentum: False

Architecture:
  model_type: rec
  algorithm: CPPD
  in_channels: 3
  Transform:
  Encoder:
    name: SVTRNet
    img_size: [32, 128]
    out_char_num: 25
    out_channels: 256
    patch_merging: 'Conv'
    embed_dim: [128, 256, 384]
    depth: [6, 6, 6]
    num_heads: [4, 8, 12]
    mixer: ['Conv','Conv','Conv','Conv','Conv','Conv', 'Conv','Conv', 'Global','Global','Global','Global','Global','Global','Global','Global','Global','Global']
    local_mixer: [[5, 5], [5, 5], [5, 5]]
    last_stage: False
    prenorm: True
  Decoder:
    name: CPPDDecoder
    vis_seq: 64
    num_layer: 2
    pos_len: False
    rec_layer: 1


Loss:
  name: CPPDLoss
  ignore_index: 100
  smoothing: True
  pos_len: False
  sideloss_weight: 1.0

PostProcess:
  name: CPPDLabelDecode
  character_dict_path: *character_dict_path
  use_space_char: *use_space_char

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: LMDBDataSet
    data_dir:  E:\Proj\PyProj\OpenOCR-main\datasets\data_cepen_3w #../Union14M-L-LMDB-Filtered
    transforms:
      - DecodeImagePIL: # load image
          img_mode: RGB
      - PARSeqAugPIL:
      - CPPDLabelEncode: # Class handling label
          pos_len: False
          character_dict_path: *character_dict_path
          use_space_char: *use_space_char
          max_text_length: *max_text_length
      - RecTVResize:
          image_shape: [32, 128]
          padding: False
      - KeepKeys:
          keep_keys: ['image', 'label', 'label_node', 'length'] # dataloader will return list in this order
  loader:
    shuffle: True
    batch_size_per_card: 256
    drop_last: True
    num_workers: 0 #4 for linux

Eval:
  dataset:
    name: LMDBDataSet
    data_dir: E:\Proj\PyProj\OpenOCR-main\datasets\data_cepen_3w\validation #../evaluation/
    transforms:
      - DecodeImagePIL: # load image
          img_mode: RGB
      - CPPDLabelEncode: # Class handling label
          pos_len: False
          character_dict_path: *character_dict_path
          use_space_char: *use_space_char
          max_text_length: *max_text_length
      - RecTVResize:
          image_shape: [32, 128]
          padding: False
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 128
    num_workers: 0 # 4 for linux

**Comment by Topdu** [comment]
If you are using a Chinese dataset, please refer to the `configs/rec/cppd/svtr_base_cppd_ch.yml` file.
Please note that ignore_index should be greater than the number of categories：
```yaml
Loss:
  name: CPPDLoss
  ignore_index: 7000 #  greater than the number of categories
  smoothing: True
  pos_len: False
  sideloss_weight: 1.0
```

**Comment by zhuzhiqiang148** [comment]
Thank you for your comment. Just to clarify, my dataset only includes digits, Latin letters, and some special symbols. It does not contain any Chinese characters.
0
1
2
3
4
5
6
7
8
9
A
B
C
D
E
F
G
H
I
G
K
L
M
N
O
P
Q
R
S
T
U
V
W
X
Y
Z
a
b
c
d
e
f
g
h
i
j
k
l
m
n
o
p
q
r
s
t
u
v
w
x
y
z
*
/
.
:
#
+
-
